misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -183,7 +182,7 @@ class MiskCommonServiceModule : KAbstractModule() {
 
 internal data class DependencyEdge(val dependent: Key<*>, val dependsOn: Key<*>)
 internal data class EnhancementEdge(val toBeEnhanced: Key<*>, val enhancement: Key<*>)
-internal data class ServiceEntry(val key: Key<out Service>)
+internal data class ServiceEntry(val key: Key<out Service>, val provider: Provider<out Service>)
```

Comment:
omit the `Provider` field and compute it on-demand?

Reply:
(that’ll be `injector.getProvider(key)` on line 133)

Reply:
fixed